pkg/internal/apis/config/validate.go

```diff
@@ -155,6 +155,11 @@ func validatePortMappings(portMappings []PortMapping) error {
 		addr := net.ParseIP(portMapping.ListenAddress)
 		addrString := addr.String()
 
+		if (portMapping.HostPort == -1 || portMapping.HostPort == 0) && (portMapping.Protocol == PortMappingProtocolTCP || portMapping.Protocol == "") {
+			// Port -1 and 0 cause a random port to be selected, thus duplicates are allowed
+			continue
+		}
```

Comment:
why the protocol check?
I think this should just go at the beginning of the loop and remove the check for protocol, no?

Reply:
I added the protocol check because the allocation logic in Kind can only allocate TCP port, but docker does support multiple random UDP ports, so you're right and the lack of support in Kind for allocating random UDP ports should probably be considered a different issue (if at all).

Reply:
I pushed a commit to move the check to the begin of the loop & drop the protocol condition